j5/boards/base.py

```diff
@@ -114,21 +113,16 @@ def __next__(self) -> Board:
         """Get the next item in the iteration."""
         if self._iterator_counter >= len(self.boards):
             raise StopIteration
-        board = self.boards[self._iterator_counter]
+        board = self.boards[list(self.boards.keys())[self._iterator_counter]]
         self._iterator_counter += 1
         return board
 
-    def __getitem__(self, index: BoardIndex) -> Board:
+    def __getitem__(self, serial: str):
         """Get the board from an index."""
         if len(self.boards) <= 0:
             raise IndexError("Could not find any boards.")
-
-        if type(index) == int:
-            return self.boards[cast(int, index)]
-        elif type(index) == str:
-            for b in self.boards:
-                if b.serial == index:
-                    return b
-            raise KeyError(f"Could not find a board with the serial {index}")
-        else:
-            raise IndexError(f"Cannot index boards with type {str(type(index))}")
+        if type(serial) != str:
+            raise KeyError("Index must be a string")
```

Comment:
This should be a `TypeError`
```suggestion
            raise TypeError("Index must be a string")
```